cli/internal/cmd/configgenerate.go

```diff
@@ -135,13 +157,40 @@ func parseGenerateFlags(cmd *cobra.Command) (generateFlags, error) {
 		return generateFlags{}, fmt.Errorf("resolving kuberentes version from flag: %w", err)
 	}
 
+	attestationString, err := cmd.Flags().GetString("attestation")
+	if err != nil {
+		return generateFlags{}, fmt.Errorf("parsing attestation flag: %w", err)
+	}
+
+	var attestationType config.AttestationType
+	// if no attestation type is specified, use the default for the cloud provider
+	if attestationString == "" {
+		attestationType = config.AttestationType("")
```

Comment:
Calling attestation type with empty string as arg is not that easy to read on the caller side, maybe add a 

```go
const AttestationTypeDefault AttestationType = "default"
```

to the constant?

It might also be a good idea to define 

```go
const AttestationTypeUnknown AttestationType = ""
```

to have an explicit nil/empty value and not implicitly assuming default where we might have forgotten to set it.